src/core/utils.js

```diff
@@ -582,3 +582,27 @@ export function addId(elem, pfx = "", txt = "", noLC = false) {
   elem.id = id;
   return id;
 }
+
+/**
+ * Returns all the descendant text nodes of an element.
+ * @param {Node} elem
+ * @param {Array:String} exclusions node types (localName) to exclude
+ * @returns {Array:String}
+ */
+export function allTextNodes(elem, exclusions) {
+  const excl = new Set(exclusions);
+  const textNodes = [];
+  getTextNodes(elem);
+  return textNodes;
+
+  function getTextNodes(node) {
```

Comment:
we probably don't need this function. Can just just inline all this.

Reply:
oh, sorry, I didn't see that it was recursive! Never mind.

Reply:
can make it iterative though

Reply:
This all still seems silly tho... would this work? 

```
function getTextNodes(el, exclusions) {
  const textNodes = [];
  const walker = document.createTreeWalker(
    el,
    NodeFilter.SHOW_TEXT,
    null,
    false
  );
  let node;
  while ((node = walker.nextNode())) {
    const isExcluded =
      exclusions.includes(node.parentElement.localName) === false;
    textNodes.push(node);
  }
  return textNodes;
}
```

Reply:
fixed typo above.

Reply:
event better!
*@sidvishnoi always forgets about TreeWalker API*

Reply:
How about: 

```JS
function getTextNodes(el, exclusions = []) {
  const acceptNode = node => {
    return exclusions.includes(node.parentElement.localName)
      ? NodeFilter.FILTER_REJECT
      : NodeFilter.FILTER_ACCEPT;
  };
  const nodeIterator = document.createNodeIterator(
    el,
    NodeFilter.SHOW_TEXT,
    { acceptNode },
    false
  );
  const textNodes = [];
  let node;
  while ((node = nodeIterator.nextNode())) {
    textNodes.push(node);
  }
  return textNodes;
}
```

Reply:
_@sidvishnoi goes to read the docs_

Reply:
Maybe the above... completely untested 🤪

Reply:
it does work.
any reason to use NodeIterator instead of TreeWalker?
``` js
  // with TreeWalker
  while (walker.nextNode()) {
    textNodes.push(walker.currentNode);
  }
```

Reply:
> NodeIterators and TreeWalkers are two different ways of representing the nodes of a document subtree and a position within the nodes they present. A NodeIterator presents a flattened view of the subtree as an ordered sequence of nodes, presented in document order. Because this view is presented without respect to hierarchy, iterators have methods to move forward and backward, but not to move up and down. Conversely, a TreeWalker maintains the hierarchical relationships of the subtree, allowing navigation of this hierarchy. In general, TreeWalkers are better for tasks in which the structure of the document around selected nodes will be manipulated, while NodeIterators are better for tasks that focus on the content of each selected node.

So, tl;dr: `NodeIterator` is flat, which is what we want.... don't worry, TIL me too!

Reply:
> someone should add this example to MDN docs! 

Thanks for volunteering! 🤣